File: development_logs/arb_ram_gen.m
width = 10;
length = 2 ^ 16;
M = 2 ^ width - 1;
t = linspace(0, pi, length / 2);
data_1 = M * 0.5 * (sin(t) + 1);
k = linspace(0, 1, length / 2);
data_2 = M * abs(k - 0.5);
data = [data_1, data_2];
plot(data);

fout = fopen('init_bram_saw_sin.coe', 'w');
fprintf(fout, 'memory_initialization_radix=10;\n');
fprintf(fout, 'memory_initialization_vector =\n');
for i = 1 : 65536
    if (i ~= 65536)
        fprintf(fout, '%.0f, ', data(i));
    else
        fprintf(fout, '%10.0f;', data(i));
    end
end
fclose(fout);

figure;
%phase_inc = 50;

File: development_logs/arb_ram_gen.py
out_file = open("init_arb_bram.coe", "w")
data = "memory_initialization_radix=10;\n"
data += "memory_initialization_vector =\n"

length = 2**16;
#print("length = %d"%(length))
data_list = {
        0 : 400,
        1 : 900,
        2 : 500,
        3 : 200,
        4 : 0,
        5 : 300,
        6 : 800,
        7 : 100
}
for i in range(0, length):
    data_i = data_list[i >> 13]
    data += str(data_i)
    if (i != length - 1):
        data += ", "

data += "; "
out_file.write(data)
out_file.close()
